src/main/java/com/basho/riak/client/core/query/timeseries/Cell.java

```diff
@@ -0,0 +1,227 @@
+package com.basho.riak.client.core.query.timeseries;
+
+import com.basho.riak.client.core.util.BinaryValue;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.Charset;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.3
+ */
+public class Cell
+{
+    public static final Charset ASCII_Charset = Charset.forName("US-ASCII");
+
+    Cell() {}
+
+    public Cell(String value)
+    {
+        this.binaryValue = BinaryValue.createFromUtf8(value);
+    }
+
+    public Cell(BinaryValue value)
+    {
+        this.binaryValue = value;
+    }
+
+    public Cell(int value)
+    {
+        this.integerValue = (long) value;
+        this.isIntegerCell = true;
+    }
+
+    public Cell(long value)
+    {
+        this.integerValue = value;
+        this.isIntegerCell = true;
+    }
+
+    public Cell(float value)
+    {
+        byte[] rawFloat = Float.toString(value).getBytes(ASCII_Charset);
+        this.numericValue = ByteBuffer.allocate(rawFloat.length).order(ByteOrder.BIG_ENDIAN).put(rawFloat).array();
+    }
+
+    public Cell(double value)
+    {
+        byte[] rawDouble = Double.toString(value).getBytes(ASCII_Charset);
+        this.numericValue = ByteBuffer.allocate(rawDouble.length).order(ByteOrder.BIG_ENDIAN).put(rawDouble).array();
+    }
+
+    public Cell(boolean value)
+    {
+        this.booleanValue = value;
+        this.isBooleanCell = true;
+    }
+
+    public Cell(Calendar value)
+    {
+        this.timestampValue = value.getTimeInMillis();
+        this.isTimestampCell = true;
+    }
+
+    public Cell(Date value)
+    {
+        this.timestampValue = value.getTime();
+        this.isTimestampCell = true;
+    }
+
+    protected BinaryValue binaryValue;
+    protected long integerValue;
+    protected byte[] numericValue;
+    protected long timestampValue;
+    protected boolean booleanValue;
+    protected byte[][] setValue;
+    protected byte[] mapValue;
+
+    private boolean isIntegerCell = false;
```

Comment:
To reduce memory consumption and increase readability it might be better to introduce CellType enum